cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperator.java

```diff
@@ -142,7 +142,6 @@ protected Future<KafkaConnectStatus> createOrUpdate(Reconciliation reconciliatio
         Map<String, String> podAnnotations = new HashMap<>();
         Map<String, String> controllerAnnotations = new HashMap<>();
 
-        boolean hasZeroReplicas = connect.getReplicas() == 0;
```

Comment:
Wat os the reason for removing this? It does not seem to be in any way relevant to the change in the code. This seems like a good way to trace all the code related to scale-to-0 problems.

Reply:
Wdym? To me this doesn't add much value, as the information is already present in KafkaConnectCluster object.

Reply:
I think it adds value -> you can trace the variable to see where it is used to see what parts to be careful about in respect to scale-to-0. The way you rewrite it is not so easy to do anymore because you have just unlinked places that check the number of replicas.